chore(docs): add mdx-embed guide

PaulieScanlon opened the pull request:
Add MDX Embed installation and links to readme

ChristianMurphy reviewed (changes requested):
Thanks @PaulieScanlon!
Could we move this to https://github.com/mdx-js/awesome? Or maybe to a guide? (maybe both)

the main readme points folks towards where to learn basics, intentionally concise, linking out to guides, rather than being comprehensive.

mdx-embed does meet the 30+ day and 50+ star requirements for https://github.com/mdx-js/awesome and would I be happy to see it as an awesome community project :sparkles:

johno commented:
Following along with @ChristianMurphy, I'd love to see this in docs/guides/mdx-embed as its own file (Which we will add nav to before v2 goes live), aside from that it looks great!

PaulieScanlon commented:
Hi @johno and @ChristianMurphy I've moved this to docs/guides/mdx-embed and added a link to the new page in docs/guides/index

Happy to make further changes if you like.

johno reviewed (approved):
Thank you @PaulieScanlon!

ChristianMurphy commented on docs/guides/mdx-embed.md:
````diff
@@ -0,0 +1,34 @@
+
+### Usage
+
+```javascript
````
```suggestion
```mdx
```
